### PR TITLE
specification: Make IDE key ownership model clearer

### LINKED ISSUE #83: [Qualcomm feedback] 7.1.2. PCIe Root Port Registration
Reference: https://lists.riscv.org/g/tech-ap-tee-io/topic/103498833#47

This model  does not seem to work when there are distrusting supervisor domains. why does the TSM have to generate and own the keys?

### PR DESCRIPTION
**Pull request by sameo**
Fixes #83

### CODE REVIEW

**Comment by sameo** [comment]
cc @ ozkoyuncu

File: specification/07-theory_operations.adoc [hunk 1]
```diff
@@ -81,10 +81,12 @@ I/O rules apply:
   used to protect the PCIe link between the TVM and the TDI.
 * A single selective IDE stream is established between the physical device and
   its Root Port. All TDIs from the physical device share this single IDE stream.
-* For each physical device from which a TDI is attached to a TVM, the TSM
-  generates, owns and distributes the IDE stream keys to both the physical
-  device and the RP.
-* The TSM configures the RP PCIe IDE extended capability.
+* For each physical device from which a TDI is bound to a TVM, the TSM that
+  manages this TVM also generates, owns and distributes the IDE stream keys to
```
Comment (jyao1) [inline specification/07-theory_operations.adoc:85]: Do we support one TSM manages the TVM, and another TSM managers the device SPDM?

Or do we mandate only one TSM must manage both TVM and the corresponding device SPDM?

**Review by jyao1 — APPROVED** [review]

**Review by rsahita — CHANGES_REQUESTED** [review]
some feedback

File: specification/07-theory_operations.adoc (hunk 1 above)
Comment (sameo) [reply specification/07-theory_operations.adoc]: @rsahita @jyao1 I tried to make that clearer with the latest version of this PR. PTAL.

File: specification/07-theory_operations.adoc [hunk 2]
```diff
@@ -81,10 +81,16 @@ I/O rules apply:
...
   its Root Port. All TDIs from the physical device share this single IDE stream.
-* For each physical device from which a TDI is attached to a TVM, the TSM
-  generates, owns and distributes the IDE stream keys to both the physical
-  device and the RP.
-* The TSM configures the RP PCIe IDE extended capability.
+* The TSM that establishes the SPDM connection with a physical device must
+  generate, own and distribute the IDE stream keys to both the physical device
+  and the RP.
+* A TSM must not bind a TVM that it manages with a TDI from a physical device
+  for which it does own the IDE keys. In other words, TDIs can only be bound to
```
Comment (rsahita) [inline specification/07-theory_operations.adoc:88]: Did you mean “does not own”? 
this sentence reads weird

**Review by rsahita — APPROVED** [review]
With some minor clarification needed

File: specification/07-theory_operations.adoc (hunk 2 above)
Comment (sameo) [reply specification/07-theory_operations.adoc]: Yes, I meant `does not own`. Does that read better now?

File: specification/07-theory_operations.adoc [hunk 3]
```diff
@@ -81,10 +81,14 @@ I/O rules apply:
...
 * A single selective IDE stream is established between the physical device and
   its Root Port. All TDIs from the physical device share this single IDE stream.
-* For each physical device from which a TDI is attached to a TVM, the TSM
-  generates, owns and distributes the IDE stream keys to both the physical
-  device and the RP.
-* The TSM configures the RP PCIe IDE extended capability.
+* The TSM that establishes the SPDM connection with a physical device must
+  generate, own and distribute the IDE stream keys to both the physical device
+  and the RP.
+* A TDI must not be bound to a TVM if the TSM managing the TVM does not owns the
```
Comment (rsahita) [inline specification/07-theory_operations.adoc:87]: Nit typo - owns -> own

**Pull request merged**